src/services/vyos-netlinkd

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env python3
+#
+# Copyright VyOS maintainers and contributors <maintainers@vyos.io>
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 2 or later as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import syslog
+import signal
+import select
+
+from pyroute2 import IPRoute # pylint: disable = no-name-in-module
+from pyroute2 import NetlinkError # pylint: disable = no-name-in-module
+from sys import exit
+
+running = True
+
+def sigterm_handler(signo, frame):
+    global running
+    running = False
+    sig = signal.Signals(signo)
+    syslog.syslog(syslog.LOG_INFO, f'Received signal {sig.name} - shutting down...')
+
+def main():
+    syslog.openlog(ident="vyos-netlinkd", logoption=syslog.LOG_PID)
+    syslog.syslog(syslog.LOG_INFO, "Netlink listener daemon started.")
+
+    ipr = IPRoute()
+    ipr.bind()
+    fd = ipr.fileno()
+
+    global running
+    while running:
+        try:
+            # Wait for up to 1 second for a netlink message
+            rlist, _, _ = select.select([fd], [], [], 1.0)
+            if not rlist:
+                # timeout - retry
+                continue
+
+            # Receive and process any messages
+            for message in ipr.get():
+                event_type = message['event']
+                match event_type:
+                    case 'RTM_NEWLINK':
+                        attrs = dict(message.get('attrs', []))
+                        iface = attrs.get('IFLA_IFNAME', '<unknown>')
+                        operstate = attrs.get('IFLA_OPERSTATE', '<unknown>')
+                        mac = attrs.get('IFLA_ADDRESS', '<unknown>')
+                        syslog.syslog(syslog.LOG_INFO,
+                                      f'RTM_NEWLINK -> {iface}, state={operstate}, mac={mac}')
```

Comment:
Will PPPoE interfaces also log?
For example, for BRAS. There could be thousands of interfaces.